hedera-node/hedera-mono-service/src/main/java/com/hedera/services/ledger/TransferLogic.java

```diff
@@ -150,19 +150,23 @@ public void doZeroSum(final List<BalanceChange> changes) {
             }
         }
 
-        if (validity == OK
-                && (autoCreationFee > 0)
-                && (autoCreationFee > (long) accountsLedger.get(txnCtx.activePayer(), BALANCE))) {
-            validity = INSUFFICIENT_PAYER_BALANCE;
-        }
-
         if (validity == OK) {
             adjustBalancesAndAllowances(changes);
             if (autoCreationFee > 0) {
-                payAutoCreationFee(autoCreationFee);
-                autoCreationLogic.submitRecordsTo(recordsHistorian);
+                // try to pay auto creation fee only **after** balances have been adjusted,
+                // so we are certain the payer has enough balance after potential transfers
```

Comment:
Amazing catch! 👏 ❤️ 

But note the rollback a few lines later,
```
    accountsLedger.undoChangesOfType(List.of(BALANCE));  
```
will leave the approval/allowance changes committed.

So it seems we need to track whether the payer balance is updated in the `BalanceChange` loop; and then compare `autoCreationFee` to that updated balance (if present) instead of `accountsLedger.get(txnCtx.activePayer(), BALANCE)`.